### PR TITLE
Feat duplicate questions

### PR DESCRIPTION
**Pull request by andela-iamadi**
This PR addresses three cards:
1: User sees disabled action buttons and a visual loader where possible when a question/answer/comment is in the process of submiting to the backend (https://trello.com/c/1l8dAewN)

2. While a user is typing in the search box, show a kind of loader(visual cue), to let the user know that autosuggestion is being done (https://trello.com/c/5bhpfGML)

3. Questions from a search should be trackable (https://trello.com/c/562bSsoa)

### CODE REVIEW

File: app/js/config/Marketing.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (abiodun0) [inline app/js/config/Marketing.js]: 🔴 Avoid module.exports

File: app/js/actions/AnswerActions.js [hunk 2]
```diff
@@ -20,11 +20,10 @@ AnswerActions = {
 
   receiveAnswers: (data) => {
     AppDispatcher.dispatch({
-      actionType: ZhishiConstants.ANSWER_INDEX,
+      actionType: ZhishiConstants.ANSWER_INDX,
```
Comment (andela-gnyenyeshi) [inline app/js/actions/AnswerActions.js]: ANSWER_INDEX vs ANSWER_INDX? I think the first one is easier to read/understand.

File: app/js/components/Home.react.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ function getHomeState(){
   let current_user = AuthStore.getCurrentUser();
 
   if (_.isEmptyObject(questions)) {
-    webAPI.processRequest(`/questions/personalized`,
+    webAPI.processRequest(/*`/questions/personalized`*/'/questions',
       "GET", {page: current_page},
```
Comment (andela-gnyenyeshi) [inline app/js/components/Home.react.js]: Single quotes for GET

File: app/js/actions/AnswerActions.js (hunk 2 above)
Comment (andela-gnyenyeshi) [inline app/js/actions/AnswerActions.js]: ANSWER_INDEX vs ANSWER_INDX which is more readable? I'd go with ANSWER_INDEX

File: app/js/components/comments/Form.react.js [hunk 4]
```diff
@@ -36,7 +37,7 @@ class NewCommentForm extends React.Component {
          <div className="field">
            <textarea style={{height:'auto'}} defaultValue={comment.content}></textarea>
          </div>
-         <button className="ui button" onClick={this.submitComment.bind(this)}>
+         <button id="submitCommentBtn" className="ui button" onClick={this.submitComment.bind(this)}>
            Post Comment
          </button>
        </form>
```
Comment (andela-gnyenyeshi) [inline app/js/components/comments/Form.react.js]: So many missing semi-colons in this file :)

File: app/js/components/questions/Question.react.js [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import React from "react"
 
+
```
Comment (andela-gnyenyeshi) [inline app/js/components/questions/Question.react.js]: Too much Whitespace.

File: app/js/components/layouts/SearchBar.react.js [hunk 6]
```diff
@@ -68,25 +69,26 @@ class SearchBar extends React.Component {
...
-   questionsToSearch() {
-    //  var questions =  SearchStore.getQuestions();
-    //  var keys = Object.keys(questions), questions_array = [], question;
-    //  keys.map(function(key) {
-    //    question = questions[key];
-    //    question = {id: question.id, title: question.title}
-    //    questions_array.push(question)
-    //  })
+   searchIcon(questions) {
+     // if the box is empty, no search is done
```
Comment (andela-gnyenyeshi) [inline app/js/components/layouts/SearchBar.react.js]: Should comments start with a lower-case letter?

File: app/js/actions/AnswerActions.js (hunk 2 above)
Comment (andela-iamadi) [inline app/js/actions/AnswerActions.js]: Lol! A typo! We should complete these test ASAP

File: app/js/components/Home.react.js (hunk 3 above)
Comment (andela-iamadi) [inline app/js/components/Home.react.js]: Huh? Hmmm. . .we should agree on that. . .

**Pull request merged**